### PR TITLE
Smol potion stuff

### PR DESCRIPTION
**Pull request by quat1024**
at the request of blayyke:

* `ThrownPotionEntity#method_7498` -> `applySplashPotion`
  * called when a splash potion lands
  * applies the splash potion effect
* `ThrownPotionEntity#method_7497` -> `applyLingeringPotion`
  * called when a lingering potion lands
  * creates an area effect cloud entity

one more:

* `LivingEntity#method_6086` -> `acceptsThrownPotions`
  * called in `applySplashPotion` and `AreaEffectCloudEntity#tick`
  * if it returns `false`, the potion effect is not applied. in vanilla armor stands have this `false`

### CODE REVIEW

File: mappings/net/minecraft/entity/LivingEntity.mapping [hunk 1]
```diff
@@ -200,6 +200,7 @@ CLASS net/minecraft/class_1309 net/minecraft/entity/LivingEntity
 	METHOD method_6085 setLivingFlag (IZ)V
 		ARG 1 mask
 		ARG 2 value
+	METHOD method_6086 canApplyPotion ()Z
```
Comment (Sollace) [inline mappings/net/minecraft/entity/LivingEntity.mapping]: I'd like to use the word `immune` somewhere in there. Anyone got a suggestion?

Comment (Pyrofab) [reply mappings/net/minecraft/entity/LivingEntity.mapping]: `isPotionImmune` ?
Edit: nevermind, this is the opposite of what it should be

Comment (quat1024) [reply mappings/net/minecraft/entity/LivingEntity.mapping]: It's reversed, entities that return false are immune

**Comment by quat1024** [comment]
Changed to `acceptsThrownPotions` through epic web editor shenanigans on my phone

File: mappings/net/minecraft/entity/LivingEntity.mapping [hunk 2]
```diff
@@ -200,6 +200,7 @@ CLASS net/minecraft/class_1309 net/minecraft/entity/LivingEntity
 	METHOD method_6085 setLivingFlag (IZ)V
 		ARG 1 mask
 		ARG 2 value
+	METHOD method_6086 acceptsThrownPotions ()Z
```
Comment (Runemoro) [inline mappings/net/minecraft/entity/LivingEntity.mapping]: "accepts" is a weird term for this. Why not `affectedBySplashPotions`?

Comment (asiekierka) [reply mappings/net/minecraft/entity/LivingEntity.mapping]: I like that suggestion., though I would debate between that and isAffectedBySplashPotions

Comment (Sollace) [reply mappings/net/minecraft/entity/LivingEntity.mapping]: `doSplashPotionsHaveAnyEffectsWhenAppliedToThisEntity()`?

Comment (Pyrofab) [reply mappings/net/minecraft/entity/LivingEntity.mapping]: True, `affectedBySplashPotions` would be a field name. It could be argued that `isAffectedBySplashPotions` is a tad too verbose, as Sollace noted, but at least it's clear.

Comment (quat1024) [reply mappings/net/minecraft/entity/LivingEntity.mapping]: I'm a bit cautious to refer to them as "splash potions" since it also affects the behavior of lingering potions, but I suppose you could consider them a type of splash potion.

**Comment by quat1024** [comment]
While `isAffectedBySplashPotions` is a bit wordy i think it's the best we've got lol

**Review by Runemoro — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]
